near-sdk/src/types/public_key.rs

```diff
@@ -60,7 +60,7 @@ impl std::str::FromStr for CurveType {
 ///             .parse()
 ///             .unwrap();
 /// ```
-#[derive(Debug, Clone, PartialEq, PartialOrd, Ord, Eq, BorshDeserialize, BorshSerialize)]
+#[derive(Debug, Clone, PartialEq, PartialOrd, Ord, Eq, BorshSerialize, Hash)]
```

Comment:
I've recently came up with a rule-of-thumb: "don't derive serialize/deserialize for newtype structs". 

For newtypes, implementing serialization by hand is easy -- just delegate to the field.  And the benefit can be large -- the "derive" part of serde/borsh is heavier than the runtime. 

For the SDK, it might makes sense to:

* check if we can get rid of all derives
* if we can, consider not depending on the corresponding proc-macros

Though, I guess for the sdk this doesn't make too much sense --- the sdk is heavy proc-macro-based itself, so there won't be any extra savings.

Reply:
yes, this can come in a future change, not going to group with this PR because it's unrelated